code/game/machinery/doors/poddoor.dm

```diff
@@ -1,381 +1,71 @@
-
 /obj/machinery/door/poddoor
 	name = "\improper Podlock"
 	desc = "That looks like it doesn't open easily."
 	icon = 'icons/obj/doors/rapid_pdoor.dmi'
 	icon_state = "pdoor1"
-	id = 1.0
-	dir = 1
 	armor = list("melee" = 50, "bullet" = 100, "laser" = 100, "energy" = 100, "bomb" = 50, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 70)
 	layer = PODDOOR_OPEN_LAYER
 	open_layer = PODDOOR_OPEN_LAYER
 	closed_layer = PODDOOR_CLOSED_LAYER
 
-/obj/machinery/door/poddoor/Initialize()
-	. = ..()
-	if(density)
-		layer = PODDOOR_CLOSED_LAYER //to override door.New() proc
-	else
-		layer = PODDOOR_OPEN_LAYER
-
 /obj/machinery/door/poddoor/Bumped(atom/AM)
 	if(!density)
 		return ..()
 	else
-		return 0
+		return FALSE
 
 /obj/machinery/door/poddoor/attackby(obj/item/W, mob/user)
 	add_fingerprint(user)
 	if(!W.pry_capable)
 		return
 	if(density && (machine_stat & NOPOWER) && !operating && !CHECK_BITFIELD(resistance_flags, INDESTRUCTIBLE))
 		spawn(0)
-			operating = 1
-			flick("pdoorc0", src)
-			icon_state = "pdoor0"
-			SetOpacity(0)
+			operating = TRUE
 			sleep(15)
-			density = 0
-			operating = 0
+			open()
```

Comment:
Any change of a `CALLBACK` here, to remove those `spawn` and `sleep` procs?

Reply:
Sure.